### ISSUE TITLE
Additional obligate macro options

### ISSUE THREAD

**Issue by vparunak** [Newbrict/obsidian-obligator#27]
It wouild be very helpful if the {{obligate}} macro supported an "every other <day, week, month, year>" option. I have some tasks that come up every other (say) Thursday, irrespective of which Thursday of the month it is. Right now I can't figure out a way to express this. This "every other" functionality would require a way to anchor the series at a fixed data, and would then enable an entry every other time period from then. Generalization: every <n> <day, week, month, year>, where "every other" would be n =2.

**Comment by Newbrict** [comment]
Hi @vparunak 

The obligate macro does support this, though it's a bit confusing I admit :^)

I have a bunch of examples of complicated obligate macros in the daily_note.md file in the example directory.

This one is probably relevant to your purposes:
```md
{{ obligate 1-7,15-21 1,3,5,7,9,11 7 }}
- [ ] every other Sunday but only every other month
```

Let me know if that helps or if you need any further assistance.

**Comment by vparunak** [comment]
If I understand the example you give, it selects the first and third Sunday of every other month. If I make the second argument *, it would be every month. But consider this series of Sundays in 2024: 1/7, 1/21, 2/4, 2/18, 3/3, 3/17, 3/31, 4/14, 4/28, ... These are all 14 days apart, but if I understand the syntax correctly, {{obligate 1-7,15-21 * 7 }} would work fine for the first six, but would miss the last three.

**Comment by Newbrict** [comment]
Yes you're right. 3/31 would not be triggered, so it would continue on 4/7. So there would be a skipped Sunday every now and then with this logic. You're also right that in order to fix that, you'd have to explicitly change your ranges for whenever that "5 Sundays in a month" event happens.

I'm hesitant to extend the syntax. If I can think of an elegant way to do that in the code I might add it. For now I'll mark this as a feature request.

**Comment by vparunak** [comment]
Here's one possible syntax: add three more fields, {{obligate <1> <2> <3> <4> <5> <6>}}
<1, 2,3> are day of month, month of year, and day of week  as before. 
<4> is calendar year. This is a natural extension of the first four, and allows scheduling obligations for specific years (e.g., your parents' 50th anniversary)
<5> is periodicity: for every second day (week, month, year) it would be 2.
<6> specifies the  unit: d(ay), w(eek), m(onth), y(ear) (or some other enumerated set)

So every second Sunday starting 1/7/24 would be
{{obligate 7 1 7 2024 2 w}}
The first four fields anchor the series, while the last two specify the repetition.

Lots of other options, of course, but I thought I'd capture this one.

**Comment by Newbrict** [comment]
Thanks for the suggestion, I'll think about it.

**Comment by marvin-te** [comment]
Hi there,
I came here because I was eyeing for such a functionality as well. But I think the implementation proposed by @vparunak is not ideal, since there already is a syntax for stepped values in cron: the slash operator.

For example:
- `*/2 * *` = "every second day of the month (aka every other day or every "even calendar day"). 
- `* */2 5` = every Friday in every other month of the year
- etc.

I think, since the obligator syntax already is a simplified cron syntax, you should stick with this approach. Cron is very powerful, widely adopted and well known (at least in the IT world). There is no reason to reinvent the wheel and confuse users who already are familiar with the cron syntax.

Other than that, thank you for your work and this neat plugin.
Cheers.
